src/drivers/imu/bmi055/BMI055_gyro.cpp

```diff
@@ -360,6 +362,18 @@ BMI055_gyro::RunImpl()
 		return;
 	}
 
+	// don't publish duplicated reads
+	if ((report.gyro_x == _gyro_prev[0]) && (report.gyro_y == _gyro_prev[1]) && (report.gyro_z == _gyro_prev[2])) {
+		perf_end(_sample_perf);
+		perf_count(_duplicates);
+		return;
+
+	} else {
```

Comment:
No `else` needed, since you have the `return`.

Reply:
Thanks, I was just trying to keep the comparison together as a single block.